### PR TITLE
MSC2175: Remove the `creator` field from `m.room.create` events

### LINKED ISSUE #1193: 'creator' field of m.room.create events is redundant
why don't we just use 'sender'?

### PR DESCRIPTION
**Pull request by richvdh**
[Rendered](https://github.com/matrix-org/matrix-doc/blob/rav/proposals/remove-creator/proposals/2175-remove-creator-field.md)

Fixes #1193

### CODE REVIEW

**Comment by KitsuneRal** [comment]
No objections from the client perspective.

**Comment by richvdh** [comment]
I'm going to assume that the only reason not to do this is tuits, so

@mscbot fcp merge

**Comment by anoadragon453** [comment]
Looks like a good cleanup, thanks.

@mscbot reviewed

**Comment by ara4n** [comment]
fwiw, the reason for the creator field was i think to support moving creator ownership around in subsequent events somehow. but given this has never been figured out, i’m happy to remove the redundancy and readd correctly jn future if needed

**Comment by turt2live** [comment]
Looks like this is still missing an implementation.

A small problem with this MSC is that one can't get the room creator from a `GET /state/m.room.create` call anymore - the endpoint only returns `content`, not the full event format. This might be more easily fixed with an MSC which changes that returned body in some capacity.

File: proposals/2175-remove-creator-field.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# MSC2175: Remove the `creator` field from `m.room.create` events
```
Comment (turt2live) [inline proposals/2175-remove-creator-field.md:1]: A potential issue with this MSC is that [`GET /state/:event/:key` ](https://matrix.org/docs/spec/client_server/r0.6.1#get-matrix-client-r0-rooms-roomid-state-eventtype-statekey) doesn't return the `sender`, just the `content`.

Comment (richvdh) [reply proposals/2175-remove-creator-field.md:1]: bleh. That mostly makes me think that `GET /state/:event/:key` is broken.

I guess that means that you can't use that API to get the creator of a room. It feels like a bit of an edge-case.

Comment (turt2live) [reply proposals/2175-remove-creator-field.md:1]: yea, I think in practice people aren't super interested in this and the clients that are will have cached it in other ways (or worst case, call `/state` to get it all)

**Comment by turt2live** [comment]
> Looks like this is still missing an implementation.
> 
> A small problem with this MSC is that one can't get the room creator from a `GET /state/m.room.create` call anymore - the endpoint only returns `content`, not the full event format. This might be more easily fixed with an MSC which changes that returned body in some capacity.

Hello ancient Travis. You were wrong. You can get the event from `GET /state`: https://github.com/matrix-org/matrix-spec/issues/1047

File: proposals/2175-remove-creator-field.md (hunk 1 above)
Comment (turt2live) [reply proposals/2175-remove-creator-field.md:1]: this thread is false: https://github.com/matrix-org/matrix-spec/issues/1047

File: proposals/2175-remove-creator-field.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+events have a mandatory `creator` property giving the ID of the user who
+created the room. This field is redundant as it is always identical to the
+`sender` of the create event.
+
+This MSC proposes that, in a future room version, this field should be removed
+and that the `sender` field be used instead.
+
+Note that `creator` is mentioned in the [auth
+rules](https://matrix.org/docs/spec/rooms/v1#authorization-rules). It can
+safely be removed.
```
Comment (richvdh) [inline proposals/2175-remove-creator-field.md:13]: I think that by "removed", we mean: "replaced with the `sender` of the `m.room.create` event".

Comment (turt2live) [reply proposals/2175-remove-creator-field.md:13]: https://github.com/matrix-org/matrix-spec-proposals/pull/3832

**Comment by turt2live** [comment]
This is now formally assigned to room version 11: https://github.com/matrix-org/matrix-spec-proposals/pull/3820

**Comment by turt2live** [comment]
Spec PR: https://github.com/matrix-org/matrix-spec/pull/1604

**Comment by turt2live** [comment]
Merged 🎉

**Comment by anoadragon453** [comment]
> _However_, as of v11, `creator` is totally unprivileged; it is under the control of the client (via `creation_content` in the `/createRoom` call). So a user could create a v11 room, and claim that someone else made it by setting `creator` to a different user.
> 
> The question is whether that matters in practice.

No special privileges are currently given to the creator of a room (other than allowing them to initially join the room), so I'm not sure what implications this would actually have. The only thing that comes to mind is moderation tooling; some tooling could attempt to clean up all rooms created by a certain user. And perhaps that tool was written before v11 rooms existed, and would be searching for `m.room.create->content->creator` across all rooms.

Still, I'm not familiar with any moderation tooling that erases rooms based on *creator*, rather than message content. Perhaps this is just a reminder that client-side tools that act on rooms should scope themselves to a set of known room versions - and warn if they find any that it does not know the rules for.

**Pull request merged**